client/src/app/auth/services/auth.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable, tap } from 'rxjs';
+import { User } from '../auth.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  constructor(private http: HttpClient) {}
+
+  login(email: string, password: string): Observable<User> {
+    return this.http
+      .post('http://localhost:3000/user/login', {
+        email,
+        password,
+      })
+      .pipe(
+        tap((res: User | any) => {
+          this.setSession(res);
+        })
+      );
+  }
+
+  private setSession(data: User): void {
+    localStorage.setItem('token', data.token);
+    localStorage.setItem(
```

Comment:
If you want to store expiresIn pls get it from API response because expiration time is set on the server using env variable